notifier/notifier.go

```diff
@@ -383,25 +391,25 @@ func (n *Notifier) Stop() {
 
 // Describe implements prometheus.Collector.
 func (n *Notifier) Describe(ch chan<- *prometheus.Desc) {
```

Comment:
@fabxc Interestingly I've encountered the opposite problem when reusing Alertmanager packages for the multi-tenant Alertmanager in Weave Cortex. The problem there is that I instantiate a complete AM chain (with notifiers, silences, etc.) for each user, but the Prometheus client library does not allow me to register the same metric name twice if I instantiate each component more than once (because those components use the non-global Collector pattern, which I otherwise really like). In that case, I noticed that global metrics would actually be better! Now I am thinking what to do: write a custom registry that concatenates metrics from different users together (but that would be high cardinality due to tracking user IDs in a label), or a custom registry that would somehow try to aggregate the metrics between users again. Meh...